### PR TITLE
Fixed issue where Provenance had null recipe if not found

### PR DESCRIPTION
**Pull request by Alos**
To keep things consistent with other fields, since a Recipe is optional, the output should just omit the field instead of adding it with `null` value.

### CODE REVIEW

File: src/main/java/io/github/intoto/slsa/models/Provenance.java [hunk 1]
```diff
@@ -25,6 +25,7 @@
    *
    * <p>MAY be unset/null if unknown, but this is DISCOURAGED.
```
Comment (adityasaky) [inline src/main/java/io/github/intoto/slsa/models/Provenance.java:26]: Should this be removed?

**Comment by adityasaky** [comment]
Otherwise, LGTM!

File: src/main/java/io/github/intoto/slsa/models/Provenance.java (hunk 1 above)
Comment (Alos) [reply src/main/java/io/github/intoto/slsa/models/Provenance.java:26]: Probably not, it illustrates to the users of the library that the field can be left unset or null, the library will just omit it.

Comment (adityasaky) [reply src/main/java/io/github/intoto/slsa/models/Provenance.java:26]: gotcha

**Pull request merged**